### PR TITLE
enforce onnx conversion in CI

### PR DESCRIPTION
**Pull request by chriselion**
### Proposed change(s)

Cherry pick of https://github.com/Unity-Technologies/ml-agents/pull/3600 to the release branch

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: ml-agents/mlagents/model_serialization.py [hunk 1]
```diff
@@ -18,6 +21,11 @@
 from tensorflow.python.framework import graph_util
 from mlagents.trainers import tensorflow_to_barracuda as tf2bc
 
+if LooseVersion(tf.__version__) < LooseVersion("1.12.0"):
```
Comment (vincentpierre) [inline ml-agents/mlagents/model_serialization.py]: Should we also filter out tf 2.x ?

File: ml-agents/mlagents/model_serialization.py [hunk 2]
```diff
@@ -67,18 +75,28 @@ def export_policy_model(
...
+                logger.exception(
+                    "Exception trying to save ONNX graph. Please report this error on "
+                    "https://github.com/Unity-Technologies/ml-agents/issues and "
+                    "attach a copy of frozen_graph_def.pb"
+                )
+
+    else:
+        if _enforce_onnx_conversion():
+            raise RuntimeError(
+                "ONNX conversion enforced, but couldn't import dependencies."
```
Comment (vincentpierre) [inline ml-agents/mlagents/model_serialization.py]: This is not true, it could also be because the version does not match.

**Comment by vincentpierre** [comment]
I realize now, this is for release, not master. 
Please merge as is and maybe add comments in another PR.

File: ml-agents/mlagents/model_serialization.py (hunk 1 above)
Comment (chriselion) [reply ml-agents/mlagents/model_serialization.py]: I don't want to explicitly filter it out. A future version of tf2onnx should support f2 2.x, in which case the `import` should succeed.

File: ml-agents/mlagents/model_serialization.py (hunk 2 above)
Comment (chriselion) [reply ml-agents/mlagents/model_serialization.py]: Good point, I'll update the comment here. It's not something a user will ever see (unless they added the environment variable), though.

**Pull request merged**